cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -89,6 +106,20 @@
     private final PersistedCassandraClusterHealthCheckHistory healthCheckHistory;
     @NotNull
     private final PersistedCassandraFrameworkConfiguration configuration;
+    @NotNull
+    private final PersistedCassandraClusterJobs jobsState;
+
+    public static int getPortMapping(CassandraFrameworkConfiguration configuration, String name) {
```

Comment:
?? Do you mean `getPortMapping` or the `ApiController` c'tor? The c'tor is really ugly - but less uglier than a singleton in `CassandraCluster`.